### DISCUSSION TITLE
Representation structure

### DISCUSSION THREAD

**Discussion by OwlyWizard** [kc-ml2/meent#76] [Q&A]
Hi,
I have gone through the tutorials, but it seems that we are not able to see the structure that we are implementing, right? I am able to see the resulting field but not the structure itself. Is there a way to show a slice of it, at a constant z for instance?
Thanks in advance!

**Comment by OwlyWizard** [comment]
It seems that there is a function draw (see paper page 40), working with vectorial representation (maybe rasterized also but it is not shown in paper). So far I cannot make it work, when I use the code below I get this error for the mee.draw function: IndexError: index 1 is out of bounds for dimension 0 with size 1

For the code, I tried to copy what is done in the paper, without success:
import torch
import meent

thickness = [300.]
length_x = 100 
length_y = 300 
center = [300, 500] 
n_index_1 = 3.48 
n_index_2 = 1 
base_n_index_of_layer = n_index_2 
angle = 35 * torch.pi / 180 
n_split = [5, 5] # degree of approximation 
 
length_x = torch.tensor( length_x , dtype = torch.float64 , requires_grad=True) 
length_y = torch.tensor( length_y , dtype = torch . float64 , requires_grad=True) 
thickness = torch.tensor( thickness , requires_grad = True ) 
angle = torch.tensor(angle , requires_grad = True )
mee = meent.call_mee(backend=2, grating_type=2, pol=0, n_I=1, n_II=1.5, theta=0, phi=0, fourier_order=30, wavelength=600, period=1000, thickness=thickness, type_complex=0, device=torch.device('cpu'), fft_type=2, improve_dft=True)

obj_list = mee.rectangle_rotate(* center , length_x , length_y , * n_split , n_index_1 , angle) 
layer_info_list = [[ base_n_index_of_layer , obj_list ]] 
mee.draw(layer_info_list)

**Reply by yonghakim** [reply]
Meent is going through intensive updates so many things are being  changed. 
Sorry for the inconvenience 😥

Can you please refer this example code?
https://github.com/kc-ml2/meent/blob/main/examples/vector_1d.py

or,

https://github.com/kc-ml2/meent/blob/main/examples/ocd/arxiv_ocd_optimize.py
This is the code of example in page 25, https://arxiv.org/pdf/2406.12904

Currently I am fully occupied for big update including Li's inverse rule and UX changes.
I plan to update documents and example code after that.

**Comment by yonghakim** [comment]
Hi, 
You are right. 
Meent doesn't support structure plot. 
I think it's good to have but the priority is low since kind of lack of manpower.

**Comment by OwlyWizard** [comment]
Thank you, I used the ocd example at the end to help me!
